e2e/test/iothub/messaging/MessageReceiveE2ETests.cs

```diff
@@ -606,21 +667,84 @@ private async Task ReceiveSingleMessageUsingCallbackAndUnsubscribeAsync(TestDevi
             }
 
             // Send a message to the device from the service.
-            (Message secondMessage, _, _) = ComposeC2dTestMessage(Logger);
-            await serviceClient.SendAsync(testDevice.Id, secondMessage).ConfigureAwait(false);
-            Logger.Trace($"Sent C2D message from service, messageId={secondMessage.MessageId} - to be received on polling ReceiveAsync");
+            (Message thirdMessage, _, _) = ComposeC2dTestMessage(Logger);
+            await serviceClient.SendAsync(testDevice.Id, thirdMessage).ConfigureAwait(false);
+            Logger.Trace($"Sent C2D message from service, messageId={thirdMessage.MessageId} - to be received on polling ReceiveAsync");
 
             // This time, the message should not be received on the callback, rather it should be received on a call to ReceiveAsync().
             Func<Task> receiveMessageOverCallback = async () =>
             {
                 await testDeviceCallbackHandler.WaitForReceiveMessageCallbackAsync(cts.Token).ConfigureAwait(false);
             };
-            Client.Message message = await deviceClient.ReceiveAsync(s_tenSeconds).ConfigureAwait(false);
-            message.MessageId.Should().Be(secondMessage.MessageId);
+            using Client.Message receivedThirdMessage = await deviceClient.ReceiveAsync(s_tenSeconds).ConfigureAwait(false);
+            receivedThirdMessage.MessageId.Should().Be(thirdMessage.MessageId);
             receiveMessageOverCallback.Should().Throw<OperationCanceledException>();
+            await deviceClient.CompleteAsync(receivedThirdMessage).ConfigureAwait(false);
 
             firstMessage.Dispose();
             secondMessage.Dispose();
+            thirdMessage.Dispose();
+
+            await deviceClient.CloseAsync().ConfigureAwait(false);
```

Comment:
Should we be disposing, not closing?

Reply:
Since we are not reusing this client again, technically both should be fine. The device client will anyway be disposed once it is out of scope.